drivers/i3c/i3c_npcx.c

```diff
@@ -2144,41 +2893,53 @@ static int npcx_i3c_init(const struct device *dev)
 		return ret;
 	}
 
+	/* Lock initial */
 	k_mutex_init(&data->lock_mutex);
 	k_sem_init(&data->sync_sem, 0, 1);
 	k_sem_init(&data->ibi_lock_sem, 1, 1);
+	k_sem_init(&data->target_lock_sem, 1, 1);
+	k_sem_init(&data->target_event_lock_sem, 1, 1);
 
 	ret = i3c_addr_slots_init(dev);
 	if (ret != 0) {
 		LOG_ERR("Addr slots init fail %d", ret);
 		return ret;
 	}
 
-	ctrl_config->is_secondary = false; /* Currently can only act as primary controller. */
-	ctrl_config->supported_hdr = I3C_MSG_HDR_DDR; /* HDR-DDR mode is supported. */
-	ctrl_config->scl.i3c = config->clocks.i3c_pp_scl_hz; /* Set I3C frequency */
+	/* Set controller default configuration */
+	config_cntlr->is_secondary = config->secondary_mode;
+	config_cntlr->supported_hdr = I3C_MSG_HDR_DDR;        /* HDR-DDR mode is supported. */
+	config_cntlr->scl.i3c = config->clocks.i3c_pp_scl_hz; /* Set I3C frequency */
 
-	ret = npcx_i3c_configure(dev, I3C_CONFIG_CONTROLLER, ctrl_config);
+	/* Initial the device as controller or target */
+	config_type = (config_target->enable | config_cntlr->is_secondary) ? I3C_CONFIG_TARGET
+									   : I3C_CONFIG_CONTROLLER;
+	config_ptr = (config_type == I3C_CONFIG_CONTROLLER) ? (void *)config_cntlr
+							    : (void *)config_target;
```

Comment:
This still doesn't make sense to me. What if config target is true and `is_secondary` is false. If it is primary controller, then it should always boot as a controller... but this will skip initializing it as a controller

Reply:
A little bit confused here.
The I3C module of the NPCX4 supports both controller and target functionality.
Before booting up the device, it can be configured to one of the following modes.
I have planned the overlay configuration as follows to correspond to the different modes:
|    | target-mode-en   |secondary  | config_type  (Result) |
|  ----  | ----  | ----  | ---- |
| Primary Controller  | false |false |I3C_CONFIG_CONTROLLER |
| Secondary Controller  | false |true |I3C_CONFIG_TARGET |
| Ordinary Target (No controller capability)  | true |false |I3C_CONFIG_TARGET |

Not sure if I understand your meaning correctly.
So if `config_target = true `and `is_secondary = false`, it means the user would like to initialize the I3C device as an ordinary target at startup.